packages/core/src/delegation/DelegationNode.ts

```diff
@@ -167,35 +167,22 @@ export class DelegationNode implements IDelegationNode {
    */
   public async getHierarchyDetails(): Promise<IDelegationHierarchyDetails> {
     if (!this.hierarchyDetails) {
-      const hierarchyDetails = await queryDetails(this.hierarchyId)
-      if (!hierarchyDetails) {
-        throw new SDKErrors.HierarchyQueryError(this.hierarchyId)
-      }
-      this.hierarchyDetails = hierarchyDetails
-      return hierarchyDetails
+      this.hierarchyDetails = await fetchDetails(this.hierarchyId)
     }
     return this.hierarchyDetails
   }
 
-  /**
-   * Fetches the parent node of this delegation node.
-   *
-   * @returns Promise containing the parent as [[DelegationNode]] or [null].
-   */
-  public async getParent(): Promise<DelegationNode | null> {
```

Comment:
This might still be relevant for externals

Reply:
Restored it in https://github.com/KILTprotocol/sdk-js/pull/648/commits/6360b90256f97d4207d2bee9764acad3c763d9d3